### PR TITLE
Add Nomad provider

### PR DESCRIPTION
**Pull request by IamTheFij**
I modeled it off the Kubernetes provider a bit. It supports setting task config at group and task levels using services and meta attributes.

### CODE REVIEW

**Comment by crazy-max** [comment]
@IamTheFij Will review this soon, thanks for your contrib!

**Comment by IamTheFij** [comment]
Just wanted to check in and see if there were any suggested changes here.

**Comment by crazy-max** [comment]
@IamTheFij It looks mostly fine to me thanks! I will create a Nomad instance on my side to play a bit with it and finalize the review.

File: docs/providers/nomad.md [hunk 1]
```diff
@@ -0,0 +1,211 @@
...
+  }
+}
+```
+
+## Configuration
+
+!!! hint
+    Environment variable `DIUN_PROVIDERS_NOMAD=true` can be used to enable this provider with default values.
+
+Default values are assigned by the Nomad client. If not provided in your Diun configuration, the client will default to using the same config values as the `noamd` cli client.
```
Comment (crazy-max) [inline docs/providers/nomad.md:92]:
```suggestion
Default values are assigned by the Nomad client. If not provided in your Diun configuration, the client will default to using the same config values as the `nomad` cli client.
```

File: docs/providers/nomad.md [hunk 2]
```diff
@@ -0,0 +1,211 @@
...
+!!! abstract "Environment variables"
+    * `DIUN_PROVIDERS_NOMAD_WATCHBYDEFAULT`
+
+## Nomad annotations
+
+You can configure more finely the way to analyze the image of your tasks through Nomad meta attributes or service tags:
+
+| Name                | Default      | Description                                                                                                                                                |
+|---------------------|--------------|------------------------------------------------------------------------------------------------------------------------------------------------------------|
+| `diun.enable`       |              | Set to true to enable image analysis of this pod                                                                                                           |
```
Comment (crazy-max) [inline docs/providers/nomad.md:202]: I think right wording is `task`, not `job` right?

```suggestion
| `diun.enable`       |              | Set to true to enable image analysis of this task                                                                                                           |
```

File: docs/providers/nomad.md [hunk 3]
```diff
@@ -0,0 +1,211 @@
...
+| `diun.regopt`       |              | [Registry options](../config/regopts.md) name to use                                                                                                       |
+| `diun.watch_repo`   | `false`      | Watch all tags of this pod image ([be careful](../faq.md#docker-hub-rate-limits) with this setting)                                                        |
```
Comment (crazy-max) [inline docs/providers/nomad.md:204]:
```suggestion
| `diun.watch_repo`   | `false`      | Watch all tags of this task image ([be careful](../faq.md#docker-hub-rate-limits) with this setting)                                                        |
```

**Review by crazy-max — CHANGES_REQUESTED** [review]
Some minor changes to fix, otherwise nice job!

**Review by crazy-max — CHANGES_REQUESTED** [review]
Can you squash your commits please? Thanks

**Review by crazy-max — APPROVED** [review]

**Pull request merged**